docs/design/fabric-v2+/fpc-cc-invocation.puml

```diff
@@ -35,74 +35,103 @@ participant "ERCC Peer1_Org1" as ERCC1 order 50
+
+  Verification is only necessary iff User is not member of Org1,
+  otherwise authenticated response enough. In that case, the user
+  also has to query different peers from different orgs in order to
+  satisfy a quorum as necessary for the lifecycle endorsement
+  policy and channel configuration changes.
+
+  NOTE: For MVP, ECCInfo only contains CC_E_PK and no ECCInfo verification is needed.
+end note
+
+CLI -> CLI : generate r_key
+note right
+    generate a symmetric key to
+    encrypt the FPC chaincode result
 end note
 
+CLI -> CLI : args_ECC1 <- encrypt <args, r_key> with CC_E_PK
 
-loop for all/"enough" endorsers
+end
 
-  User1 -> User1 : args_ECC1 <- encrypt args with ECC1_E_PK
+group invoke
+  CLI -> Peer ++: send proposal <CC_NAME, args_ECC1,....>
 
-  User1 -> ECC1 ++: send proposal (args_ECC1)
+  alt FPC Chaincode is not running
+      note right Peer
+          handled implicitly by the peer which determines
+          when to call ‘run’ from external builder.
+      end note
 
-  group invoke
+      create participant "ECC" as ECC1 order 26
+      Peer -> ECC1 ++: call external builder `run`
+      note right
+          NOTE: Any failure in ECC or FPC_Enclave will cause
+          ecc (and docker) dieing so peer will notice it.
+      end note
 
-    ECC1 -> Enclave1 ++: check enclave is running
-    return
 
-    alt enclave does not exists
       ECC1 -> Manager1 ++: get sealed state
-      note right Manager1 : get from local storage
+      note right Manager1
+        ECC manager has access to peer local
+        storage through external builder.
+      end note
       return
+
+      create participant "FPC_Enclave" as Enclave1 order 30 #99FF99
       ECC1 -> Enclave1 ++ : start enclave with sealed state
       Enclave1 -> Enclave1 : unseal state and restore
       Enclave1 <-> TLCC1: establish attested secure channel
-      note over TLCC1
-        protocol is SIGMA with 1-way local attestation of TLCC
-        with reportbody = <CH_id, TLCC_inst_id> where
-        TLCC_inst_id as unique (randomly generated) identifier of
-        this particular TLCC __instance__
-      end note
-      note over Enclave1
-        Verifies local attestation against TLCC_MR and CH_id
-        from sealed state. Remembers TLCC_inst_id.
+      note right Enclave1
+          protocol is SIGMA with 1-way local attestation of TLCC
+          with reportbody = <CH_id, TLCC_inst_id> where
+          TLCC_inst_id as unique (randomly generated) identifier of
+          this particular TLCC __instance__.
+
+          Verifies local attestation against TLCC_MR and CH_id
+          from sealed state. Remembers TLCC_inst_id.
       end note
       return
-    else
-      |||
+      return
+  end
 
-      ECC1 -> Enclave1 ++: invoke (proposal(args_ECC))
+      Peer -> ECC1 ++: invoke (proposal(args_ECC), CTX)
+      note right
+        Establish context (CTX) used for all interactions
```

Comment:
Maybe calling this "transaction context" with acronym "TCtx" or "TX-CTX" or alike would make the purpose of this more explicit

Reply:
> Maybe calling this "transaction context" with acronym "TCtx" or "TX-CTX" or alike would make the purpose of this more explicit

Transaction context makes sense and I updated the note! I would like to keep ctx as it is nice and short.